packages/campaigns/components/CampaignForm/index.jsx

```diff
@@ -79,13 +79,19 @@ const colors = [
 /* Component */
 const CampaignForm = ({
   translations,
-  onCreateCampaignClick,
+  onCreateOrUpdateCampaignClick,
   onCancelClick,
   isSaving,
+  inEditMode,
+  campaignDetails,
 }) => {
   // State
-  const [campaignName, setName] = useState('');
-  const [colorSelected, setColor] = useState(purple);
+  const defaultName = campaignDetails?.name ?? '';
+  const defaultColor = campaignDetails?.color ?? purple;
+  const campaignId = campaignDetails?.id ?? null;
+  const orgId = campaignDetails?.globalOrganizationId ?? null;
```

Comment:
Hey @chayjay I noticed you are creating a few variables here for campaignDetails, I think a nice refactor would be deconstructing these into a single one, something like this:

```suggestion
const {
    id: campaignId,
    globalOrganizationId: orgId = null,
    name: defaultName = '',
    color: defaultColor = purple,
} = campaignDetails ?? {};
```